### PR TITLE
Fix for plugin installation via web interface

### PR DESCRIPTION
**Pull request by SchrodingersGat**
- Previous implementation did not correctly catch error output
- Refactored into new file plugin.installer.py
- Provides better "output" when plugin does install correctly

Ref: https://github.com/inventree/InvenTree/issues/5645

### CODE REVIEW

**Comment by SchrodingersGat** [comment]
@matmair LMK what you think of this one - this is the *original* bug I was trying to fix.

File: InvenTree/plugin/installer.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+def uninstall_plugin(plugin_name):
+    """Uninstall a plugin from the python virtual environment.
+
+    Note: This is intended to be implemented in a future version
+
+    - Run pip uninstall (only if plugin is installed in venv)
+    - Reverse any migrations for plugins which implement the app mixin
+    - Raise error if located in "plugins" or external plugin dir
+    """
+    ...
```
Comment (matmair) [inline InvenTree/plugin/installer.py:239]: Is this intended to be here?

**Review by matmair — APPROVED** [review]

File: InvenTree/plugin/installer.py (hunk 1 above)
Comment (SchrodingersGat) [reply InvenTree/plugin/installer.py]: No, was a placeholder that I missed. Thanks :)

**Review by matmair — APPROVED** [review]
Lgtm

**Pull request merged**